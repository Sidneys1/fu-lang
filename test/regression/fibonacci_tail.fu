fib: u64(n: u64) => _fib(0, 1, n);

_fib: u64(a: u64, b: u64, n: u64) = {
  if (n < 1) return a;
  if (n == 1) return b;
  return _fib(b, a + b, n - 1);
};

main: u64() => fib(90); // 2,880,067,194,370,816,120
