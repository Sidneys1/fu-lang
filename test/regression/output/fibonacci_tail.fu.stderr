INFO:fu.compiler:Output binary `a.out`: 1.07 KiB
